src/repository.rs

```diff
@@ -317,11 +321,18 @@ where
         HttpRepository {
             url: url,
             client: client,
+            http_hook: None,
             user_agent: user_agent,
             _interchange: PhantomData,
         }
     }
 
+    /// Set an hook function to be invoked before sending an HTTP request.
+    pub fn set_http_hook(&mut self, hook: Option<HttpHook>) -> &Self {
```

Comment:
* I also don't like getters&setters, but I thought the non-prefixed one is typically used for getters not for setters. But I will amend it.
 * Sure. Is something similar to the test-case (minus the panic part) ok?